### DISCUSSION TITLE
/articles/sharepoint-add-in-migration-faq/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#226] [Comments]
Customers are starting to ask about migrating SharePoint Add-ins after Microsoft announced retirement plans. In this article, I answer some common questions.
https://www.andrewconnell.com/articles/sharepoint-add-in-migration-faq/

**Comment by msterlingjones** [comment]
How do we detect existing SPD Workflows and especially Add-ins in use in our tenant?

**Reply by andrewconnell** [reply]
Do you mean how can you find all instances of your SharePoint 2013 workflows or SharePoint Add-ins deployed & in use within your tenant?

**Reply by andrewconnell** [reply]
To the best of my knowledge, there's no easy OOTB way to look across your entire tenant... you have to look at each site collection (yeah... a farm walk  🤷‍♂️ 🫤). From there, you could use the PnP PowerShell, specifically the [Get-PnPSharePointAddIn](https://pnp.github.io/powershell/cmdlets/Get-PnPSharePointAddIn.html) cmdlet to find them.

As for SharePoint 2013 workflows... again, there's no easy way to do this, but I found this same question that has a PowerShell option that might be helpful: [PowerShell Code to Find SharePoint 2013 and 2010 Workflows](https://learn.microsoft.com/en-us/answers/questions/111871/powershell-code-to-find-sharepoint-2013-and-2010-w).

**Comment by msterlingjones** [comment]
Great presentation today by the way in todays Viva Connections Community Call.

**Reply by andrewconnell** [reply]
Thanks!

**Comment by OverkillTV** [comment]
So does that mean that if I need to extract the items from a list inside SharePoint hosted Add-Ins (isolated domain) I can do it through the WebPartManager object ?

**Reply by andrewconnell** [reply]
Not really...

- The **WebPartManager** on a page is only useful for getting data stored with the web part.
- The data within a list (regardless if it's an app web or normal SharePoint site) is accessible via the SharePoint REST API or via Microsoft Graph.

For example, if you have a web part that displays products stored in a list, you need to use the SharePoint REST API/Microsoft Graph to get the products from the SharePoint list. If that web part had a public property that let the user pick the featured product to display via the property pane & stored the product's ID it in a public property on the web part, you'd use the **WebPartManager** to get the product's ID from that public property.

**Comment by nallara** [comment]
Hi,
We have a provide hosted Add-in, which used to deploy commands(custom menus based document selection) which redirects to different website with standard tokens. Is this model is supported In SPFx based apps?

**Reply by andrewconnell** [reply]
If you're referring to the context menu (aka: ECB) on a item in a list, or in the toolbar of a list, then yes, you can do that with SPFx. Those are called **ListView Command Sets**... it's a type of SPFx Extension.

Is that what you're looking for? If so, I do cover those in my course: [Mastering the SPFx](https://www.voitanos.io/course-master-sharepoint-framework/).

These command set extensions are meant to replace the older `<customAction>` element, but you only have access to those two areas (not the other options like we had outside of SPFx in the classic UX mode, such as Site Actions or the Site Settings pages).

**Reply by nallara** [reply]
Thanks for reply. I will go through  your "Mastering the SPFx" page.
Can we use still standard tokens while making call to other Website or RESTAPI?

**Reply by andrewconnell** [reply]
@nallara said:
> Can we use still standard tokens while making call to other Website or RESTAPI?

I'm not sure what you mean by **standard tokens**. All calls to SharePoint's REST API or Microsoft Graph uses OAuth access tokens. But, the SharePoint Framework API contains helpers to call SharePoint Online (SPO) & Microsoft Graph.

**Reply by nallara** [reply]
In our existing add-in(provider hosted add-in) command bar control action, we are calling a different site and loading data. When we are redirecting/calling external link we are sending StandardTokens in the form of query strings.

CommandUIHandler Command="LoadData"                  
CommandAction="https://testsite.com/Pages/load.aspx?{StandardTokens}&amp;Item={SelectedItemId}&amp;List={SelectedListId}

When command action executed we get StandardTokens like SPHostUrl, language, productnumer etc. this info can be found at https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/url-strings-and-tokens-in-sharepoint-add-ins#tokens-that-you-can-use-inside-a-url.

The StandardTokens combines five other tokens, those are SPHostUrl={HostUrl}&amp;SPAppWebUrl={AppWebUrl}&amp;SPLanguage={Language}&amp;SPClientTag={ClientTag}&amp;SPProductNumber={ProductNumber}

I want to know whether still we can call external resource/website using with StandardTokens options as commandAction in SPFx based app.

**Reply by andrewconnell** [reply]
@nallara said:
> I want to know whether still we can call external resource/website using with StandardTokens options as commandAction in SPFx based app.

I honestly have no idea, but I doubt it. You'd have to test it (but it would be pretty easy to test). Nothing in SPFx will replace those, plus the `AppWebUrl` one has nothing to do with the modern UX and I doubt the SPO client-side stuff would replace them.

The recommended way to do this in a modern approach would be to either use Microsoft Graph to get data from other sites OR to use search (via Microsoft Graph) and apply a filter for the specified sites.
